### PR TITLE
fix: Remove optional chaining where applicable

### PR DESCRIPTION
**Pull request by traeok**
## Proposed changes

This PR fixes a couple of variable accesses that use optional chaining, even though we use these variables (earlier in the code) without optional chaining.

Resolves 2 issues detected by Coverity scans.

## Release Notes

Milestone: 2.13.0

Changelog: None

## Types of changes

What types of changes does your code introduce to Zowe Explorer?
_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This checklist will be used as reference for both the contributor and the reviewer_

- [x] I have read the [CONTRIBUTOR GUIDANCE](https://github.com/zowe/vscode-extension-for-zowe/wiki/Best-Practices:-Contributor-Guidance) wiki
- [x] PR title follows [Conventional Commits Guidelines](https://www.conventionalcommits.org/en/v1.0.0-beta.2/)
- [x] PR Description is included
- [ ] gif or screenshot is included if visual changes are made
- [x] `yarn workspace vscode-extension-for-zowe vscode:prepublish` has been executed
- [x] All checks have passed (DCO, Jenkins and Code Coverage)
- [ ] I have added unit test and it is passing
- [ ] I have added integration test and it is passing
- [ ] There is coverage for the code that I have added
- [ ] I have tested it manually and there are no regressions found

### CODE REVIEW

**Review by t1m0thyj — DISMISSED** [review]
LGTM, thanks @traeok! The change to `shared/utils.ts` may be rendered unnecessary by #2564, but that should be an easy conflict to address in a future merge 🙂

**Comment by traeok** [comment]
@JillieBeanSim sorry for the confusion, I've rebased off of `main` and should be ready for re-review.

File: packages/zowe-explorer/src/shared/utils.ts
Comment (rudyflores) [inline packages/zowe-explorer/src/shared/utils.ts:302]: I see this portion missing some code coverage, should we write a test for it or is this fine since it seemed to be passing before the removal of optional chaining? @traeok

**Review by rudyflores — CHANGES_REQUESTED** [review]

File: packages/zowe-explorer/src/shared/utils.ts
Comment (rudyflores) [reply packages/zowe-explorer/src/shared/utils.ts:302]: I see, so i'm assuming once that is merged, it should also fix your breaking check for codecov?

Comment (traeok) [reply packages/zowe-explorer/src/shared/utils.ts:302]: Yep, I think we can let this PR sit until #2564 is merged and `main` is updated with the latest changes from `maintenance`. Once the changes are on main, I'll pick his incoming changes during the merge commit and that should resolve the patch coverage 🙂

**Review by t1m0thyj — DISMISSED** [review]

**Review by t1m0thyj — COMMENTED** [review]
Will approve after changes from #2580 have been merged

File: packages/zowe-explorer/src/dataset/actions.ts
Comment (zFernand0) [inline packages/zowe-explorer/src/dataset/actions.ts:1070]: I always get worried when we change the signature of an exported function. 😨 
But if others are ok with this, I think it's fine 😋

**Review by zFernand0 — APPROVED** [review]
Left a small comment 😅 
LGTM! 😋

File: packages/zowe-explorer/src/dataset/actions.ts
Comment (JillieBeanSim) [reply packages/zowe-explorer/src/dataset/actions.ts:1070]: I feel you @zFernand0 
I think this will be fine since it isn't extender facing or part of the ZE API package.

**Review by t1m0thyj — APPROVED** [review]
LGTM, thanks @traeok!

**Review by JillieBeanSim — APPROVED** [review]

**Pull request merged**
